pydantic/types.py

```diff
@@ -89,9 +89,12 @@
 
 
 if TYPE_CHECKING:  # pragma: no cover
+    from .dataclasses import DataclassType  # noqa: F401
+    from .main import BaseModel  # noqa: F401
     from .utils import AnyCallable
 
     CallableGenerator = Generator[AnyCallable, None, None]
+    ModelType = Optional[Type[Union['BaseModel', 'DataclassType']]]
```

Comment:
This name is deceptive, better to have

```py
ModelOrDc = Type[Union['BaseModel', 'DataclassType']]
```

So the name is clear.

Reply:
Done, but set to:

```ModelOrDc = Optional[Type[Union['BaseModel', 'DataclassType']]]```

Reply:
Maybe better to create without Optional? And then just write Optional[ModelOrDc] where it is used? What do you think? I can easily make the change if required.

Reply:
Yes please, then it's clear.

Reply:
Done!